fluent_pose_synthesis/data/load_data.py

```diff
@@ -0,0 +1,203 @@
+
+        total_clips = len(self.clip_indices)
+        total_frames = sum(frame_nums_list)
+        print('Dataset loaded, trained with %d clips, %d frames, %d mins in total' % (total_clips, total_frames, total_frames/25/60))   # All sequences are sampled at 25fps
+
+    def __len__(self) -> int:
+        """
+        Returns the total number of sliding-window clips in the dataset.
+        """
+        return len(self.clip_indices)
+
+    def __getitem__(self, idx: int) -> Dict[str, Any]:
+        """
+        Retrieves a sliding-window clip from the dataset.
+        Args:
+            idx (int): Index of the clip to retrieve.
+        """
+        # Each row of clip_indices: [example_index, frame0, frame1, ..., frame(window_size-1)]
+        indices = self.clip_indices[idx]
+        ex_idx = indices[0]
+        # The remaining indices are the frame indices for this clip
+        clip_frame_indices = indices[1:]
+        example = self.examples[ex_idx]
+        
+        # Slice the pose sequences using the precomputed indices
+        disfluent_seq = example['disfluent'].body.data[clip_frame_indices].astype(self.dtype)
+        fluent_seq = example['fluent'].body.data[clip_frame_indices].astype(self.dtype)
+        
+        input_clip = disfluent_seq[:self.disfluent_frame]
+        target_clip = fluent_seq[self.disfluent_frame:]
+                
+        # Get the mask for the fluent pose sequence over these frames
+        mask = example['fluent'].body.mask[clip_frame_indices]
+        
+        return {
+            'data': target_clip,
+            'conditions': {
+                'input_clip': input_clip,
+                'mask': mask,
+                'metadata': example['metadata']
+            }
+        }
+
+
+def custom_collate(batch):
+    """
+    Custom collate function to handle the metadata field properly.
+    Stacks the numerical arrays while merging metadata dicts into one dict, where each key maps to a list of values from the batch.
+    """
+    collated = {}
+    # Stack the 'data' field (target clips)
+    collated['data'] = np.stack([item['data'] for item in batch])
+    conditions = {}
+    # Stack input_clip and mask (both are NumPy arrays of equal shape)
+    conditions['input_clip'] = np.stack([item['conditions']['input_clip'] for item in batch])
+    conditions['mask'] = np.stack([item['conditions']['mask'] for item in batch])
+    
+    # Merge metadata (each is a dict) into a single dict where each key maps to a list of values
+    merged_metadata = {}
+    for item in batch:
+        meta: dict = item['conditions']['metadata']
+        for key, value in meta.items():
+            if key not in merged_metadata:
+                merged_metadata[key] = []
+            merged_metadata[key].append(value)
+    conditions['metadata'] = merged_metadata
+    
+    collated['conditions'] = conditions
+    
+    return collated
+
+
+def test_dataset():
```

Comment:
very nice! would love to run it.
maybe the name can be a bit better: `example_dataset` (anything that starts with `test` usually is a test (to be run with `pytest` for example)